mission_control/navigator_missions/nav_missions/coral_survey.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+from __future__ import division
+
+import txros
+import tf
+import tf.transformations as trns
+import numpy as np
+import navigator_tools
+from navigator_tools import fprint
+from navigator_singleton import pose_editor
+from twisted.internet import defer
+
+WEST = trns.quaternion_matrix(pose_editor.WEST)
+EAST = trns.quaternion_matrix(pose_editor.EAST)
+NORTH = trns.quaternion_matrix(pose_editor.NORTH)
+SOUTH = trns.quaternion_matrix(pose_editor.SOUTH)
+
+@txros.util.cancellableInlineCallbacks
+def main(navigator):
+    result = navigator.fetch_result()
+
+    middle_point = np.array([-10, -70, 0]) #yield navigator.database_query("coral_survey")
+
+    quads_to_search = [1, 2, 3, 4]
+    if (yield navigator.nh.has_param("/mission/coral_survey/quadrants")):
+        quads_to_search = yield navigator.nh.get_param("/mission/coral_survey/quadrants")
+
+    waypoint_from_center = np.array([10 * np.sqrt(2)])
+
+    # Construct waypoint list along NSEW directions then rotate 45 degrees to get a good spot to go to.
+    directions = [EAST, NORTH, WEST, SOUTH]
+    waypoints = []
+    for quad in quads_to_search:
+        mid = navigator.move.set_position(middle_point).set_orientation(directions[quad - 1])
+        waypoints.append(mid.yaw_left(45, "deg").forward(waypoint_from_center).set_orientation(NORTH))
+
+    # Get into the coral survey area
+    yield waypoints[0].go()
+
+    # Publish ogrid with boundaries to stay inside
+    ogrid = OgridFactory(middle_point, draw_borders=True)
+    msg = ogrid.get_message()
+    latched = navigator.latching_publisher("/mission_ogrid", OccupancyGrid, msg)
+
+    searcher = navigator.search("coral_survey", waypoints)
+    yield searcher.start_search(move_type='skid', spotings_req=1)
+
+    fprint("Centering over the thing!", title="CORAL_SURVEY")
+
+    # TODO: Center over the thing.
+
+    boat_position = (yield navigator.tx_pose)[0]
+
+    # TODO: make this easier
+    quad = np.argmin(np.linalg.norm(boat_position - [[w.pose[0][0][0], w.pose[0][1][0],w.pose[0][2][0]] for w in waypoints], axis=1)) + 1
```

Comment:
Not sure why `pose_editor` is storing the pose in such a contorted way such that I have to index it like this